### PR TITLE
VS Code 1.59.0 Extension API

### PR DESCRIPTION
**Pull request by rzhao271**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/microsoft/vscode/blob/release/1.59/src/vs/vscode.d.ts
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/vscode/index.d.ts [hunk 1]
```diff
@@ -10927,7 +10929,7 @@ declare module 'vscode' {
          * @param scope A scope for which the configuration is asked for.
          * @return The full configuration or a subset.
          */
-        export function getConfiguration(section?: string, scope?: ConfigurationScope | null): WorkspaceConfiguration;
+        export function getConfiguration(section?: string | undefined, scope?: ConfigurationScope | null): WorkspaceConfiguration;
```
Comment (nicolas377) [inline types/vscode/index.d.ts:10932]: you don't need to add undefined to a optional param

File: types/vscode/index.d.ts [hunk 2]
```diff
@@ -11547,7 +11549,7 @@ declare module 'vscode' {
         /**
          * The metadata of this cell. Can be anything but must be JSON-stringifyable.
          */
-        readonly metadata: { [key: string]: any };
+        readonly metadata: { [key: string]: any }
```
Comment (nicolas377) [inline types/vscode/index.d.ts:11552]: you have to have a semicolon here

**Review by nicolas377 — CHANGES_REQUESTED** [review]
these two things are what's causing the CI to fail

**Comment by nicolas377** [comment]
Also, you should probably be updating tests somewhere, since this is a breaking change

**Comment by rzhao271** [comment]
The API is being tested here https://github.com/microsoft/vscode/tree/release/1.59/extensions/vscode-api-tests

**Comment by rzhao271** [comment]
I also don't seem to have write perms yet. This PR was created automatically by a CI action, but I'll fork the repo and create another one.

**Pull request closed without merging**